### PR TITLE
Pattern Assembler: Add the activation screen

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/77483

## Proposed Changes

* Moved `NavigatorProvider` outside the `StepContainer` so we're able to get the navigator to navigate between screens
* Added the Activation notice to notify their homepage will be replaced
* Added the Activation screen, and clicking the “Continue” button will enter this screen
* Display the “Back” button on the Activation screen

**Screenshot**

**Demo**

https://github.com/Automattic/wp-calypso/assets/13596067/6e1e05fc-d066-464e-b4b6-3653fefa9234

## Testing Instructions

* Use Calypso Live link [below](https://github.com/Automattic/wp-calypso/pull/77475#issuecomment-1566532503)
* Go to `/themes`
* Scroll down to select BCPA CTA
* On the Pattern Assembler screen
  * Select layout and styles
  * Continue
  * Ensure you're able to see the Activation screen
  * Click the “Back” button on the top-left corner, and ensure you're able to go back to the main screen
  * Continue again
  * Click the checkbox and the “Activate“ button
  * Ensure you're able to land on the site editor and see the selected layout and styles

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx [hunk 1]
```diff
@@ -176,6 +183,26 @@ const ScreenMain = ( {
 							</NavigationButtonAsItem>
 						</>
 					</NavigatorItemGroup>
+					{ ! isNoticeDismissed && (
+						<Notice className="screen-container__notice" onRemove={ dismissNotice }>
+							{ translate(
+								'After activation, your homepage content will be replaced. But you can still access your old content. {{a}}Learn more{{/a}}.',
```
Comment (okmttdhr) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx]: badge

Am I correct in understanding what actually will happen is that "your _pages/posts_ will persist, but your _contents in any templates_ will be deleted"?

If so,

>After activation, your homepage content will be replaced. But you can still access your old content. {{a}}Learn more{{/a}}.

Wouldn't this copy mislead people into thinking that all content is preserved? (I'm sorry If I'm wrong 🙏 )

**Comment by miksansegundo** [comment]
Can you share a link to understand the context of these changes? They work fine but I'd like to understand the motivation for them.

**Comment by arthur791004** [comment]
@miksansegundo Here - pbxlJb-3RH-p2#comment-2569

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx (hunk 1 above)
Comment (arthur791004) [reply client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx]: > but your contents in any templates will be deleted

Not exactly. The content on the previous theme will be preserved, so they can switch back to the previous theme and their existing homepage will come back. However, if the existing theme is the BC3, then yes, all the content will be replaced with the new one.

**Comment by taipeicoder** [comment]
Code looks good to me, but I have a design feedback 🙂
See: pbxlJb-3RH-p2#comment-2597.

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx (hunk 1 above)
Comment (okmttdhr) [reply client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx]: I now understand what will happen. Thank you!

**Review by okmttdhr — APPROVED** [review]
Tested and LGTM (other than https://github.com/Automattic/wp-calypso/pull/77475#issuecomment-1566899717) ✅

I just want to note that I never saw the activation screen when I directly accessed the Site Assembler (`/setup/site-setup/patternAssembler?siteSlug=<existing-site-with-theme>`). Somehow related to this issue https://github.com/Automattic/wp-calypso/issues/77490?

**Review by taipeicoder — APPROVED** [review]
LGTM 👍
The design feedback is addressed in pbxlJb-3RH-p2#comment-2598

**Comment by arthur791004** [comment]
Update: Remove the activation notice as it's a bit duplicate. See Tv3pYqA3EcRfiXC31IxrXE-fi-535:17443

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8016209

Thank you @arthur791004 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8016209).

**Pull request merged**